File: d3/p2/p2.py
import re
intxt = 'in.txt'
with open(intxt, 'r') as file:
    text = file.read()
matches = re.findall(r"mul\(\d{1,3},\d{1,3}\)|don't\(\)|do\(\)", text)
sum = 0
enabled = True
for statement in matches:
    if statement == "do()":
        enabled = True
    elif statement == "don't()":
        enabled = False
    elif enabled: 
        sum += int(statement.split(",")[0][4:]) * int(statement.split(",")[1][:-1])
print(sum)

File: d3/p2/p2.php
<?php
$text = file_get_contents('in.txt');
preg_match_all('/mul\(\d{1,3},\d{1,3}\)|don\'t\(\)|do\(\)/', $text, $matches);
$sum = 0;
$enabled = true;  
foreach ($matches[0] as $match) {
    if ($match == 'do()') {
        $enabled = true;
    }
    elseif ($match == "don't()") {
        $enabled = false;
    }
    elseif ($enabled) {
        $sum += (int)(substr(explode(",", $match)[0], 4)) * (int)(substr(explode(",", $match)[1], 0, -1));
    }
}
echo $sum;
?>